openedx/openedx-platform
# Dashboard announcements feature

This PR adds an Announcement feature to show announcements on the sidebar of the LMS dashboard. It uses TinyMCE to show an editor in the studio.

**JIRA tickets**: Implements OC-3201

**Screenshots**:

![dashboard announcements](https://user-images.githubusercontent.com/7556571/32713023-53721626-c897-11e7-9514-0abbdb8a89ad.png)
![screenshot-2018-4-24 edit announcements studio](https://user-images.githubusercontent.com/270966/39393694-4314a4f2-4a7d-11e8-9ed2-e60bb83f8c27.png)
![screenshot-2018-4-24 edit announcement studio](https://user-images.githubusercontent.com/270966/39393705-52feb8c6-4a7d-11e8-886b-4c0063154aac.png)
![screenshot-2018-4-24 delete announcement studio](https://user-images.githubusercontent.com/270966/39393703-50623fde-4a7d-11e8-889f-c8ff02c92770.png)


**Sandbox URL**:

* LMS: https://pr16496.sandbox.opencraft.hosting/
* Studio: https://studio-pr16496.sandbox.opencraft.hosting/

**Testing setup**:

1. Enable announcements with `settings.FEATURES['ENABLE_ANNOUNCEMENTS'] = True`.
1. Run migrations `./manage.py cms migrate --settings=devstack`

**Testing instructions**:

1. Login to Studio as a staff member.
1. Navigate to the announcements page in studio: https://studio-pr16496.sandbox.opencraft.hosting/maintenance/announcements/
1. Click the "Create New", add content, ensure announcement is enabled
1. Add more than 5 announcements to check pagination.
1. Login to the LMS as a student.
1. View the Course Dashboard and check that the Announcement is properly displayed
1. Edit the Announcement, disable the active field, and ensure that it is removed from the LMS Dashboard.
1. Delete an announcement and check that the delete endpoint works.

**Author notes and concerns**:

1. I wrote tests for the JSX component using [enzyme](https://github.com/airbnb/enzyme) and [jest](https://facebook.github.io/jest/docs/en/tutorial-react.html), similar to the ones in [Paragon](https://github.com/edx/paragon/blob/master/src/Tabs/Tabs.test.jsx), but I was unable to run them without adding a number of dependencies in package.json which seem out of the scope of this pull request. Are there already plans for adding JSX testing into edx-platform? If desired I could open another PR with those dependencies and configuration options.

**Reviewers**

**Settings**
```yaml
EDXAPP_FEATURES:
  ENABLE_ANNOUNCEMENTS: true
```
